src/cdomains/intDomain.ml

```diff
@@ -266,7 +266,15 @@ struct
       let ur = if Int64.compare max_int x2 = 0 then y2 else x2 in
       norm @@ Some (lr,ur)
 
+  (* Returns top if one of a, b is top, otherwise return result *)
+  let on_top_return_top_bin a b result =
```

Comment:
The way it is implemented, this swallows `ArithmeticOnIntegerBot` exceptions in case where one of the arguments is `\top` and the other one is `\bot`. Is this by design?

Reply:
Actually, it doesn't.